feat: add new structure of `telegram` notifier

mohammadhonarvar opened the pull request.

mohammadhonarvar committed:
refactor: add the basic structure from `nanolib`

mohammadhonarvar committed:
refactor(telegram): fix errors & complete the structure

mohammadhonarvar committed:
chore(.vscode): update

mohammadhonarvar committed:
deps: update

mohammadhonarvar committed:
refactor(telegram): sending message & the process of `start` command

mohammadhonarvar committed:
refactor(telegram): types

mohammadhonarvar committed:
fix(telegram): start `bot`

mohammadhonarvar committed:
feat(telegram): add `new_group` command to create a group

mohammadhonarvar committed:
fix(telegram): remove creating a group as default when it doesn't exist

mohammadhonarvar committed:
fix(telegram): validate `groupId`

mohammadhonarvar committed:
feat(telegram): add `cryptoFactory`

mohammadhonarvar committed:
fix(telegram): add a missing `import` & format

mohammadhonarvar committed:
fix(telegram): read `dropPendingUpdates` from `env` object

mohammadhonarvar committed:
fix: update `scripts`

mohammadhonarvar committed:
fix(telegram): types

mohammadhonarvar committed:
fix(telegram): check the access to run the command

mohammadhonarvar committed:
feat(telegram): add `getUnsubscribeInlineKeyboardData` to reuse

mohammadhonarvar committed:
refactor(telegram): keys of `message` & fix all its usages

mohammadhonarvar committed:
fix(config): add `botAdminChatId`

mohammadhonarvar committed:
feat: add a `demo.http`

mohammadhonarvar committed:
docs: update

alimd commented on .github/dependabot.yml:
add docker plz

mohammadhonarvar replied:
Hi there.
I'm sorry for not getting back to you sooner.
I've added `deploy` directory and I've added `Dockerfile` in `telegram` directory.

Please check the `env` in `deploy`.

alimd replied:
منظورم همین package-ecosystem: docker بعد
مرسی
همون ویس تو گروه راحت تره
درضمن there هم خودتی

mohammadhonarvar committed:
feat: add `deploy` directory

mohammadhonarvar committed:
feat(telegram): add `Dockerfile`

mohammadhonarvar committed:
feat: add `.github` configs based on `weaver-starter-kit`

mohammadhonarvar committed:
refactor(deploay): add some changes based on `weaver-starter-kit`

mohammadhonarvar committed:
chore: update `lerna` based on `weaver-starter-kit`

mohammadhonarvar committed:
refactor(packages): rename `telegram` to `notifier-api`

mohammadhonarvar committed:
deps: update

mohammadhonarvar committed:
fix(.github): some issues

mohammadhonarvar committed:
fix(deploy): `env` values

mohammadhonarvar committed:
refactor(notifier-api): rename all `group`s to `category`

mohammadhonarvar committed:
refactor(notifier-api): copy path in `Dockerfile`

mohammadhonarvar committed:
refactor(notifier-api): rename all `group`s to `category`

mohammadhonarvar committed:
feat(notifier-api): use `AlwatrHashGenerator`

mohammadhonarvar committed:
feat(notifier-api): use `hashGenerator` & save data

mohammadhonarvar committed:
feat(notifier-api): start the bot & handle errors

mohammadhonarvar committed:
chore(notifier-api): cleanup

mohammadhonarvar committed:
refactor(notifier-api): add some pre-handlers & complete the process of `notify` route & cleanup codes

alimd commented on .github/workflows/codeql.yaml:
```diff
@@ -6,13 +6,13 @@ on:
-    #        │ │ └───────── day of the month (1 - 31)
-    #        │ └─────────── hour (0 - 23)
-    #        └───────────── minute (0 - 59)
+    - notifier-api: '0 7 * */2 *'
```
خدایی چرا اسم این عوض شده؟! 😄 اینجوری که کار نمیکنه

alimd reviewed (changes requested):
مرسی
تغییرات رو یا خودم انجام میدم یا در گروه ویس میدم

alimd committed:
feat(notifier-api): Add parseBodyAsJson handler

This commit adds the parseBodyAsJson handler to the notifier-api package. The handler is responsible for parsing the request body as JSON and handling any errors that occur during the process.

The handler checks if the request body is empty and returns an error response if it is. It also attempts to parse the body as JSON and sets the parsed body in the sharedMeta object. If the parsing fails, an error response is returned.

This commit improves the functionality of the notifier-api package by adding the parseBodyAsJson handler, which enhances the handling of JSON request bodies.

alimd committed:
refactor(vscode): Update .vscode/settings.json

alimd committed:
refactor(notifier-api): Update parseBodyAsJson handler to use JsonObject type

alimd committed:
refactor(notifier-api): rewrite notify route

alimd committed:
refactor(notifier-api): Update type definitions for Category, Member, and NotifyOption

alimd committed:
refactor: rename files and cleaup

alimd committed:
refactor(notifier-api): Update start-command to add user to category members list

alimd committed:
refactor(notifier-api): Update notify-validation handler to validate category and message inputs

alimd committed:
refactor(notifier-api): Update bot start method and error handling

alimd committed:
refactor(notifier-api): Update escapeMessage function to use regex for character escaping

alimd committed:
refactor(notifier-api): Update i18n message function and resource keys

alimd committed:
refactor(notifier-api): Update notifyTelegram function to send messages to all members of a category

alimd committed:
refactor(notifier-api): Update notifyRoute to use notifyValidation and notifyTelegram

This commit updates the notifyRoute in the notifier-api package. It replaces the previous validation logic with the new notifyValidation handler and updates the message sending logic to use the notifyTelegram function. This improves the overall functionality and maintainability of the code.

alimd committed:
refactor(notifier-api): Update config.ts to use telegramBot instead of bot

alimd committed:
refactor(notifier-api): Update home route to return JSON response

alimd committed:
refactor(notifier-api): Update import path for config in server.ts

alimd committed:
refactor(notifier-api): Update import path for nitrobase in nitrobase.ts

alimd committed:
refactor(notifier-api): rename nitrobase

alimd committed:
refactor(notifier-api): Remove unused hashGenerator

alimd committed:
refactor(notifier-api): Rename package to "telegram-notifier"

alimd committed:
docs(notifier-api): Rename package to "telegram-notifier"

alimd committed:
refactor(notifier-api): Refactor bot.ts and main.ts

- Refactor bot.ts to extract the bot start logic into a separate function called startBot()
- Update main.ts to import the startBot() function and call it to start the bot
- Update import paths in main.ts to match the renamed files and directories

alimd committed:
refactor: review and refactor everything (#243)
